spec/controllers/projects_controller_spec.rb

```diff
@@ -1,13 +1,85 @@
 require 'spec_helper'
 
+RSpec.shared_examples 'denies access' do
+  before { request }
+
+  it 'redirects to the root path' do
+    expect(response).to redirect_to(root_path)
+  end
+
+  it 'sets the flash' do
+    expect(flash[:alert]).to eq('You are not authorized to do that')
+  end
+end
+
+RSpec.shared_examples 'can update project' do
```

Comment:
This doesn't seem as if it should be a shared example, its very coupled to the `update` action on the project controller. I think it would be better to put it there.